Add CI workflow

samherrmann opened the pull request:
With the addition of this workflow, the following code changes were made:

- Replace math/rand with crypto/rand

    From staticcheck: math/rand.Read has been deprecated since Go 1.20
    because it shouldn't be used: For almost all use cases,
    crypto/rand.Read is more appropriate. (SA1019)

- Add documentation comment to package

    From staticcheck: at least one file in a package should have a
    package comment (ST1000)

samherrmann commented:
My goal is to deliver this CI workflow before working on #2 so that there can be some confidence that code changes do not break existing tests.

zaninime commented:
Thanks a lot for the contribution! Very minor comment, however if you had time to split the commits it with proper names it would be great. At the moment there's only one commit that includes all the changes.

samherrmann committed:
Replace math/rand with crypto/rand

From staticcheck: math/rand.Read has been deprecated since Go 1.20
because it shouldn't be used: For almost all use cases crypto/rand.Read
is more appropriate. (SA1019)

samherrmann committed:
Add documentation comment to package

From staticcheck: at least one file in a package should have a package
comment (ST1000)

samherrmann committed:
Upgrade Go version

Go only support the last two major versions. This commit is to ensure
that this project uses a supported version of Go.

Reference:
https://go.dev/doc/devel/release#policy

samherrmann committed:
Add GitHub CI Workflow

samherrmann commented:
I hope the changes I made is what you had in mind, otherwise don't be shy to request further changes. 🙂

zaninime commented:
These are really on-spot, can't request more really 😉 I already spotted some work around the ACCM framing in your fork, happy to merge it in once you're ready. Thanks again for contributing!
